internals/plan/plan.go

```diff
@@ -1085,19 +1199,85 @@ func (p *Plan) checkCycles() error {
 }
 
 func ParseLayer(order int, label string, data []byte) (*Layer, error) {
-	layer := Layer{
-		Services:   map[string]*Service{},
-		Checks:     map[string]*Check{},
-		LogTargets: map[string]*LogTarget{},
-	}
-	dec := yaml.NewDecoder(bytes.NewBuffer(data))
-	dec.KnownFields(true)
-	err := dec.Decode(&layer)
+	layer := &Layer{
+		Services:   make(map[string]*Service),
+		Checks:     make(map[string]*Check),
+		LogTargets: make(map[string]*LogTarget),
+		Sections:   make(map[string]LayerSection),
+	}
+
+	// The following manual approach is required because:
+	//
+	// 1. Extended sections are YAML inlined, and also do not have a
+	// concrete type at this level, we cannot simply unmarshal the layer
+	// in one step.
+	//
+	// 2. We honor KnownFields = true behaviour for non extended schema
+	// sections, and at the top field level, which includes Section field
+	// names.
+	builtinSections := map[string]interface{}{
+		"summary":     &layer.Summary,
```

Comment:
Could you take a look what I did in the last commit to solve this ? Please let me know if this could work, or if you have a different proposal. I chose a solution that also uses reflection in the unit test.

Reply:
https://github.com/canonical/pebble/pull/488/commits/d3ae02cf61a972f2ea377ab2b8e8af6f55b85d90

Reply:
Resolving.